lib/aws-record/record/batch.rb

```diff
@@ -0,0 +1,53 @@
+# Copyright 2015-2016 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You may not
+# use this file except in compliance with the License. A copy of the License is
+# located at
+#
+#     http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is distributed on
+# an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+# or implied. See the License for the specific language governing permissions
+# and limitations under the License.
+
+module Aws
+  module Record
+    class Batch
```

Comment:
Name-wise, this could be confusing when we get to integration of `Aws::DynamoDB::Client#batch_get_item`. It appears this is only usable for writing, so a name like `WriteBatch` could be more appropriate.